### PR TITLE
GrafanaUI: Add success state to ClipboardButton

### PR DESCRIPTION
**Pull request by evictorero**
**What this PR does / why we need it**:
This PR starts as a suggestion from UX that in the `ClipboardButton` the feedback of the action is too far from where the action is happening in Public Dashboards.
We are showing a toast notification at the upper right corner of the screen every time a user copies a link to a clipboard.
We collect some feedback from other teams and it seems we do not have a convention for doing this so this PR is an attempt to standardize this functionality across all Grafana.

Changes inside the `ClipboardButton` component
* Added feedback inside the button changing the text by `Copied!`
* Added feedback inside the button changing icon by `check` along the text

Changes from the usages of ClipboardButton
* Removed all appNotifications (toasts) from the callers
* Changed color to primary for most cases when it makes sense. Maybe we should do it in all of them?
* Removed nested Icon component and use the property with 'copy' icon to improve UI padding
* In the Snapshot sharing tab: 
    * Changed link and color button in favor of input and blue button for consistency
**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise-partnerships-team/issues/269

**Special notes for your reviewer**:
Is it necessary to be explicit about “copy to clipboard” or we should change all of those by just “copy”?
I think is not ideal to change the size of the button after clicking it but this PR could be the start of consolidating all uses into something. We could iterate this.

I put some screenshots [in this document](https://docs.google.com/document/d/1KucZf7gEAh_UGrThxjscAbjtRmWCpnlgoC8tflj-b68/edit?usp=sharing) with the old and new UI so it is easier to see and compare the differences.

### CODE REVIEW

**Review by owensmallwood — APPROVED** [review]
Nice 🎉

File: packages/grafana-ui/src/components/Table/TableCellInspectModal.tsx [hunk 1]
```diff
@@ -60,15 +59,8 @@ export function TableCellInspectModal({ value, onDismiss, mode }: TableCellInspe
...
-        <ClipboardButton getText={() => text} onClipboardCopy={() => setIsInClipboard(true)}>
-          {!isInClipboard ? (
-            'Copy to Clipboard'
-          ) : (
-            <>
-              <Icon name="check" />
-              Copied to clipboard
-            </>
-          )}
+        <ClipboardButton icon="check" getText={() => text} onClipboardCopy={() => setIsInClipboard(true)}>
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Table/TableCellInspectModal.tsx:62]: I don't think this needs the `onClipboardCopy={() => setIsInClipboard(true)}` and more (and the associated code from it) as that was doing the same "Copied!" stuff that you've added in the base component

**Review by joshhunt — COMMENTED** [review]
Great stuff, very thorough! 

I'm not sure if the interaction is the best, with the button changing size after copied. Will discuss this in daily sync and see what we want to do (especially within the scope of _your_ PR)

File: packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx [hunk 2]
```diff
@@ -12,9 +12,16 @@ export interface Props extends ButtonProps {
...
+  buttonProps.icon = buttonIcon;
+
   const buttonRef = useRef<null | HTMLButtonElement>(null);
   const copyTextCallback = useCallback(async () => {
     const textToCopy = getText();
+    setButtonText('Copied!');
+    setButtonIcon('check');
 
     try {
       await copyText(textToCopy, buttonRef);
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx:27]: Also, we should set the success state once we actually successfully copy text - after the `await copyText` call!

File: public/app/features/dashboard/components/ShareModal/ShareLink.tsx [hunk 3]
```diff
@@ -99,7 +94,7 @@ export class ShareLink extends PureComponent<Props, State> {
         <FieldSet>
           <Field
             label="Lock time range"
-            description={isRelativeTime ? 'Transforms the current relative time range to an absolute time range' : ''}
+            description={isRelativeTime && 'Transforms the current relative time range to an absolute time range'}
```
Comment (ivanortegaalba) [inline public/app/features/dashboard/components/ShareModal/ShareLink.tsx:97]: Could this be rendered `"null"` or `"false"` strings if `isRelativeTime` is a falsy value?

Comment (joshhunt) [reply public/app/features/dashboard/components/ShareModal/ShareLink.tsx:97]: this actually shouldn't be here at all!!!!!! unrelated change i was testing out (to answer your exact question 😅

File: packages/grafana-ui/src/components/Table/TableCellInspectModal.tsx [hunk 4]
```diff
@@ -60,15 +42,8 @@ export function TableCellInspectModal({ value, onDismiss, mode }: TableCellInspe
...
-          {!isInClipboard ? (
-            'Copy to Clipboard'
-          ) : (
-            <>
-              <Icon name="check" />
-              Copied to clipboard
-            </>
-          )}
+        <ClipboardButton icon="copy" getText={() => text}>
+          Copy to Clipboard
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Table/TableCellInspectModal.tsx:46]: Thanks for updating this piece of code!

**Review by dprokop — APPROVED** [review]
Works and looks great code-wise! 

My only comment is around the copy of the button when the user clicked it. I wouldn't be too strict about not changing the width of the button due to the copy change. IMO as long as the button doesn't change its position, subtle width change should be just fine.

**Pull request merged**